sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -839,41 +855,39 @@ private synchronized void showUpdateDialog() {
         AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(mForegroundActivity);
         dialogBuilder.setTitle(R.string.mobile_center_distribute_update_dialog_title);
         final ReleaseDetails releaseDetails = mReleaseDetails;
-        String releaseNotes = releaseDetails.getReleaseNotes();
-        if (TextUtils.isEmpty(releaseNotes))
-            dialogBuilder.setMessage(R.string.mobile_center_distribute_update_dialog_message);
-        else
-            dialogBuilder.setMessage(releaseNotes);
+        String message;
+        if (releaseDetails.isMandatoryUpdate()) {
+            message = mContext.getString(R.string.mobile_center_distribute_update_dialog_message_mandatory);
+        } else {
+            message = mContext.getString(R.string.mobile_center_distribute_update_dialog_message_optional);
+        }
+        message = String.format(message, getAppName(), mReleaseDetails.getShortVersion(), mReleaseDetails.getVersion());
```

Comment:
You can make `getInstallReadyMessage` method taking a resource ID as a parameter and then use it.